File: Day - 11 (2D Arrays)/try.py
def main():
    arr = [[0] * 6 for _ in range(6)]

    # Input
    for i in range(6):
        arr[i] = list(map(int, input().split()))

    max_sum = 0

    # Calculate hourglass sum
    for i in range(4):
        for j in range(4):
            temp_sum = 0

            # top row
            temp_sum += arr[i][j]
            temp_sum += arr[i][j + 1]
            temp_sum += arr[i][j + 2]

            # middle
            temp_sum += arr[i + 1][j + 1]

            # bottom row
            temp_sum += arr[i + 2][j]
            temp_sum += arr[i + 2][j + 1]
            temp_sum += arr[i + 2][j + 2]

            # if first hourglass, set as max
            if temp_sum > max_sum or (i == 0 and j == 0):
                max_sum = temp_sum

    # Output
    print(max_sum)

if __name__ == "__main__":
    main()

File: Day - 11 (2D Arrays)/try.java
import java.util.Scanner;

public class HourglassSum {

    public static void main(String[] args) {
        int[][] arr = new int[6][6];
        Scanner scanner = new Scanner(System.in);

        // Input
        for (int i = 0; i < 6; i++) {
            for (int j = 0; j < 6; j++) {
                arr[i][j] = scanner.nextInt();
            }
        }

        int maxSum = 0;
        int tempSum;

        // Calculate hourglass sum
        for (int i = 0; i < 4; i++) {
            for (int j = 0; j < 4; j++) {
                tempSum = 0;

                // top row
                tempSum += arr[i][j];
                tempSum += arr[i][j + 1];
                tempSum += arr[i][j + 2];

                // middle
                tempSum += arr[i + 1][j + 1];

                // bottom row
                tempSum += arr[i + 2][j];
                tempSum += arr[i + 2][j + 1];
                tempSum += arr[i + 2][j + 2];

                // if first hourglass, set as max
                if (tempSum > maxSum || (i == 0 && j == 0)) {
                    maxSum = tempSum;
                }
            }
        }

        // Output
        System.out.println(maxSum);
    }
}
